File: ios/Plugin/CapacitorPluginsPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorPluginsPlugin, "CapacitorPlugins",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(log, CAPPluginReturnPromise);
)

File: ios/Plugin/CapacitorPluginsPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorPluginsPlugin)
public class CapacitorPluginsPlugin: CAPPlugin {
    private let implementation = CapacitorPlugins()
    
    private let logPlugin = LoggerPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func log(call: CAPPluginCall) {
        if let options = call.options {
            call.resolve(["params": logPlugin.log(params: options)])
        } else {
            print("no params!")
        }
    }
}
